### ISSUE TITLE
Make Flow ID header configurable

### ISSUE THREAD

**Issue by wndhydrnt** [zalando/skipper#3024]
**Is your feature request related to a problem? Please describe.**
The company I'm working for uses a different header for flow logs, for historical reasons.
The field `flow-id` in the access logs of skipper is always empty due to this.
Being able to link access logs with other data through the ID would be helpful while analyzing requests.

**Describe the solution you would like**
Ideally, it is possible to set the header via a flag passed to the skipper binary on startup. Something like `--flow-id-header`.

**Describe alternatives you've considered (optional)**
Change the internal tooling to use the `X-Flow-Id` header instead. That would be a huge effort that currently seems not worth it.

**Additional context (optional)**
N/A

**Would you like to work on it?**
Yes.
Could start in CW 19.

**Comment by AlexanderYastrebov** [comment]
As a workaround you may try adding default filters like so:
```
-default-filters-prepend='copyRequestHeader("X-Flux-Id", "X-Flow-Id") -> dropHeader("X-Flux-Id")'
-default-filters-append='copyRequestHeader("X-Flow-Id", "X-Flux-Id") -> dropHeader("X-Flow-Id")'
```

**Comment by wndhydrnt** [comment]
Tested the workaround today. It correctly sets the new header, `X-Flux-Id`, but the field `flow-id` in the access logs stays empty.
My guess is that this happens because the write of the access log is deferred [here](https://github.com/zalando/skipper/blob/ac9e8527186cea4efd9b16a44a76da76ee8d856c/proxy/proxy.go#L1553) and therefore happens after all filters have been processed already.

While looking into this, I noticed that it is possible to add custom fields to the access log through a filter via the `AccessLogAdditionalDataKey` of the state bag. A custom filter can set its own value of the field `flow-id`, which is an acceptable workaround for this issue.

Closing the issue for now.
